<launch>
	<arg name="robot_name" default="robot2"/>
	
    <node name="$(var robot_name)_amcl_node" pkg="nav2_amcl" exec="amcl" output="screen">
				<param name="base_frame_id" value="$(var robot_name)/base_footprint"/>
				<param name="odom_frame_id" value="$(var robot_name)/odom"/>
				<param name="scan_topic" value="/$(var robot_name)/scan"/>
				<remap from="/tf_static" to="tf_static"/>
				<remap from="/tf" to="tf"/>
				<remap from="/initialpose" to="/$(var robot_name)/initialpose"/>
				<remap from="/tf_static" to="tf_static"/>
				<param name="global_frame_id" value="map" />
				<param name="map_topic" value="/map"/>
				
	</node>	
		
	<node name="$(var robot_name)_lifecycle_amcl_node" pkg="nav2_lifecycle_manager" exec="lifecycle_manager" output="screen">
				<param name="autostart" value="true"/>
				<param name="use_sim_time" value="false"/>
				<param name="node_names" value="[$(var robot_name)_amcl_node]"/>
	</node>
</launch>
